hoot-core/src/main/cpp/hoot/core/io/OsmWriter.cpp

```diff
@@ -301,13 +311,20 @@ void OsmWriter::_writeNodes(shared_ptr<const OsmMap> map, QXmlStreamWriter& writ
       writer.writeEndElement();
     }
 
-    if (_includeDebug)
+    if (_includeDebug && hasStatus == false)
     {
       writer.writeStartElement("tag");
       writer.writeAttribute("k", "hoot:status");
       writer.writeAttribute("v", QString("%1").arg(n->getStatus().getEnum()));
       writer.writeEndElement();
     }
+    else if (_textStatus && n->getTags().getNonDebugCount() > 0 && hasStatus == false)
+    {
+      writer.writeStartElement("tag");
+      writer.writeAttribute("k", "hoot:status");
+      writer.writeAttribute("v", QString("%1").arg(n->getStatus().toTextStatus()));
```

Comment:
Same comment as above. The ```QString("%1").arg(...)``` isn't needed to build a string, because ```.toTextStatus()``` already returns a string.